### PR TITLE
Log in

### PR DESCRIPTION
**Pull request by jengeb**
* log in and sign up page with flexboxes 
* mobile CSS
* form validation
* http posts to backend

### CODE REVIEW

**Comment by nschloe** [comment]
When specifying email/pw, indications that the pw is too short or the mail address invalid is only given after when box is unselected. At least for the pw, this should probably be fixed. (Assuming that the mail address is always entered first.)

**Comment by andrenarchy** [comment]
We could enable validation directly and add `ng-model-options="{debounce: 500}"` to the password field. Then the validation kicks in with a delay of 500 milliseconds. @jengeb do you want to try that?

**Comment by nschloe** [comment]
The icon alignment in the buttons is a little off.

align

Not sure if this is an issue.

**Comment by jengeb** [comment]
Yes I'll work on that.

**Comment by andrenarchy** [comment]
@jengeb I merged the `returnPathService` in `authService` and fixed a few links (log in/sign in links on other pages).

File: app/controllers/login.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      $scope.hasError = function(field) {
+        const form = $scope.loginForm;
+        return (form.$submitted || form[field].$touched) &&
+          form[field].$invalid;
+      };
+
+      $scope.$watch('emailOrUsername', function() {
+        $scope.emailError = undefined;
+        $scope.responseError = undefined;
+      });
```
Comment (andrenarchy) [inline app/controllers/login.ts]: Do we really need these `$watch` statements? `emailError` is not set anywhere so it seems to be safe to remove (also from the html). `responseError` is be set to undefined in `login()` and this seems to be sufficient.

Comment (jengeb) [inline app/controllers/login.ts]: Alright, @andrenarchy do you remove it?

Comment (andrenarchy) [inline app/controllers/login.ts]: yep!

File: less/paperhive/login.less [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  display: flex;
+  -webkit-flex-direction: column;
+  flex-direction: column;
+}
+
+.ph-inner-flex-item {
+  -webkit-align-self: center;
+  align-self: center;
+  -webkit-flex: 1;
+  flex: 1;
```
Comment (andrenarchy) [inline less/paperhive/login.less]: In firefox the vertical rule is not displayed and the spacing is wrong:

screenshot from 2016-02-10 23-18-26

Removing this `flex: 1` fixes the issue (and it still works in chrome). I'm no flexbox expert so I'm not aware of the consequences. @jengeb: do we need this? :)

Comment (jengeb) [inline less/paperhive/login.less]: Oh, I didn't check firefox...

`flex` defines the length of the flex item relative to the other items inside the `flex-container`. So I thought we need this. But if it also works without this CSS code in chrome, please remove it.

**Comment by andrenarchy** [comment]
I suggest we first merge this PR. The CSS can be improved after the merge.

@nschloe what do you think?

**Comment by nschloe** [comment]
We need to merge so the other branches can go ahead.

**Comment by andrenarchy** [comment]
@nschloe then go ahead! :)

**Pull request merged**